Source/Evolve/WarpXEvolve.cpp

```diff
@@ -226,6 +212,24 @@ WarpX::Evolve (int numsteps)
             ComputeSpaceChargeField( reset_fields );
         }
 
+        if ((cur_time + dt[0] >= stop_time - 1.e-3*dt[0]) ||
+            (step == numsteps_max-1) ||
+            (synchronize_velocity_for_diagnostics &&
+                (multi_diags->DoComputeAndPack(step) ||
+                 reduced_diags->DoDiags(step)))) {
```

Comment:
The above `if` condition looks like should not modify the default behavior of the code (which is probably what we want). However, it seems that a lot of automated tests are failing because of `checkSum` differences. @dpgrote Do you understand why?

Reply:
Also, could you explain why this block of code was moved here?

Reply:
Hi @RemiLehe - I moved the block of code to the end of the loop so that the state of the data when the half v push is done is the same as it will be at the start of the next step when the half push will be undone. Ideally, they should exactly cancel each other. This is particularly important when running with electrostatic where the push needs to be done after the new fields are computed. It seems that this block should be here anyway, since the field gather done in the PushP should be done after the particle boundary conditions are applied.

This would explain why some of the CI tests are failing, that something was changed between where this block was and where it is now. I'll look into it to understand what is happening.